packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -512,8 +512,8 @@ export abstract class BrowserCrawler<
                 options: enqueueOptions,
                 page,
                 requestQueue: await this.getRequestQueue(),
-                originalRequestUrl: new URL(crawlingContext.request.url).origin,
-                finalRequestUrl: new URL(crawlingContext.request.loadedUrl ?? crawlingContext.request.url).origin,
+                originalRequestUrl: crawlingContext.request.url,
+                finalRequestUrl: crawlingContext.request.loadedUrl,
```

Comment:
Pretty big change so I hope it does not break something else 😅

Reply:
Those two are used only for relative link resolution, where we need full URL, so I believe this should be safe. We explicitly use just the origin from them in the `resolveBaseUrlForEnqueueLinksFiltering` method.